### PR TITLE
rpc: Add new descriptive fields to ListChannels RPC

### PR DESCRIPTION
**Pull request by orbitalturtle**
As described in #1639, this PR adds a few new fields to each channel listed in the ListChannels RPC.   

This is my first PR to lnd. More than happy to make changes :-) :-p.

### CODE REVIEW

File: rpcserver.go [hunk 1]
```diff
@@ -1773,6 +1773,11 @@ func (r *rpcServer) ListChannels(ctx context.Context,
 		localBalance := localCommit.LocalBalance
 		remoteBalance := localCommit.RemoteBalance
 
+                // Also for display purposes, we'll subtract the channel's "reserve"
+                // to show many satoshis can actually be spent
```
Comment (halseth) [inline rpcserver.go]: Indentation is off. Try running `make fmt`.

**Comment by orbitalturtle** [comment]
Fixed formatting

File: lnrpc/rpc.swagger.json [hunk 2]
```diff
@@ -894,7 +937,7 @@
     },
     "/v1/transactions": {
       "get": {
-        "summary": "* lncli: `listchaintxns`\nGetTransactions returns a list describing all the known transactions\nrelevant to the wallet.",
+        "summary": "* lncli: `unlock`\nUnlockWallet is used at startup of lnd to provide a password to unlock\nthe wallet database.",
```
Comment (ranchalp) [inline lnrpc/rpc.swagger.json]:
```suggestion
         "summary": "* lncli: `listchaintxns`\nGetTransactions returns a list describing all the known transactions\nrelevant to the wallet.", 
```

File: lnrpc/rpc.swagger.json [hunk 3]
```diff
@@ -909,7 +952,7 @@
         ]
       },
       "post": {
-        "summary": "* lncli: `sendcoins`\nSendCoins executes a request to send coins to a particular address. Unlike\nSendMany, this RPC call only allows creating a single output at a time. If\nneither target_conf, or sat_per_byte are set, then the internal wallet will\nconsult its fee model to determine a fee for the default confirmation\ntarget.",
+        "summary": "* lncli: `changepassword`\nChangePassword changes the password of the encrypted wallet. This will\nautomatically unlock the wallet database if successful.",
```
Comment (ranchalp) [inline lnrpc/rpc.swagger.json]:
```suggestion
         "summary": "* lncli: `sendcoins`\nSendCoins executes a request to send coins to a particular address. Unlike\nSendMany, this RPC call only allows creating a single output at a time. If\nneither target_conf, or sat_per_byte are set, then the internal wallet will\nconsult its fee model to determine a fee for the default confirmation\ntarget.", 
```

File: lnrpc/rpc.swagger.json [hunk 4]
```diff
@@ -34,7 +35,8 @@
     },
     "/v1/balance/channels": {
       "get": {
-        "summary": "* lncli: `channelbalance`\nChannelBalance returns the total funds available across all open channels\nin satoshis.",
+        "summary": "* \nInitWallet is used when lnd is starting up for the first time to fully\ninitialize the daemon and its internal wallet. At the very least a wallet\npassword must be provided. This will be used to encrypt sensitive material\non disk.",
```
Comment (ranchalp) [inline lnrpc/rpc.swagger.json]:
```suggestion
         "summary": "* lncli: `channelbalance`\nChannelBalance returns the total funds available across all open channels\nin satoshis.", 
```

File: lnrpc/rpc.swagger.json [hunk 5]
```diff
@@ -34,7 +35,8 @@
...
+        "description": "In the case of a recovery scenario, the user can also specify their aezeed\nmnemonic and passphrase. If set, then the daemon will use this prior state\nto initialize its internal wallet.\n\nAlternatively, this can be used along with the GenSeed RPC to obtain a\nseed, then present it to the user. Once it has been verified by the user,\nthe seed can be fed into this RPC in order to commit the new wallet.",
```
Comment (ranchalp) [inline lnrpc/rpc.swagger.json]:
```suggestion

```

File: lnrpc/rpc.swagger.json [hunk 6]
```diff
@@ -17,7 +17,8 @@
   "paths": {
     "/v1/balance/blockchain": {
       "get": {
-        "summary": "* lncli: `walletbalance`\nWalletBalance returns total unspent outputs(confirmed and unconfirmed), all\nconfirmed unspent outputs and all unconfirmed unspent outputs under control\nof the wallet.",
+        "summary": "*\nGenSeed is the first method that should be used to instantiate a new lnd\ninstance. This method allows a caller to generate a new aezeed cipher seed\ngiven an optional passphrase. If provided, the passphrase will be necessary\nto decrypt the cipherseed to expose the internal wallet seed.",
```
Comment (ranchalp) [inline lnrpc/rpc.swagger.json]:
```suggestion
         "summary": "* lncli: `walletbalance`\nWalletBalance returns total unspent outputs(confirmed and unconfirmed), all\nconfirmed unspent outputs and all unconfirmed unspent outputs under control\nof the wallet.", 
```

File: lnrpc/rpc.swagger.json [hunk 7]
```diff
@@ -17,7 +17,8 @@
...
+        "description": "Once the cipherseed is obtained and verified by the user, the InitWallet\nmethod should be used to commit the newly generated seed, and create the\nwallet.",
```
Comment (ranchalp) [inline lnrpc/rpc.swagger.json]:
```suggestion

```

**Comment by ranchalp** [comment]
Somebody with write access can have a look at this?

**Comment by halseth** [comment]
@ranchalp You shouldn't manually change autogenerated code like that.

**Comment by wpaulino** [comment]
Closing in favor of #3238.

**Pull request closed without merging**